auction9-frontend/src/app/data-table/data-table.component.ts

```diff
@@ -29,32 +30,24 @@ export interface AuctionData {
 })
 
 export class DataTableComponent implements AfterViewInit {
+  // data from parent
+  @Input() auctionsArray : AuctionData[];
 
   displayedColumns: string[] = ['auctionID', 'title', 'price', 'action'];
   dataSource: MatTableDataSource<AuctionData>;
 
   @ViewChild(MatPaginator) paginator: MatPaginator;
   @ViewChild(MatSort) sort: MatSort;
 
-  constructor(private auctionService: AuctionService, private route: ActivatedRoute) { }
+  constructor(private auctionService: AuctionService, private cdRef: ChangeDetectorRef) { }
 
   ngAfterViewInit() {
-    // save current path. example '/auctions'
-    const currentPath = this.route.snapshot['_routerState'].url;
-    // on '/auctions' path, get only active auctions
-    if (currentPath === '/auctions') {
-      this.auctionService.getActiveAuctions().then((data: []) => {
-      const auctions = Array.from(data);
-      this.dataSource = new MatTableDataSource(auctions);
-      
-      this.dataSource.paginator = this.paginator;
-      this.dataSource.sort = this.sort;
-      });
-    } else if (currentPath === '/myAuctions') {
-      // get auctions for this user
-    } else if (currentPath === '/wonAuctions') {
-      // get won auctions by user
-    }
+    const auctions = Array.from(this.auctionsArray);
```

Comment:
Same as previous comment, use some generic variable name, now it's auctions, but in some other page, it can be for example users. We do not want component to be aware which data contains. As I mentioned above, name of the variable can be `data` or something that has more sense.

Reply:
Renamed to generic name 'dataTable'